src/gluonnlp/data/dataset.py

```diff
@@ -78,6 +78,79 @@ def _corpus_dataset_process(s, bos, eos):
         tokens.append(eos)
     return tokens
 
+class TSVDataset(SimpleDataset):
+    """Common tab separated text dataset that reads text fields based on provided sample splitter
+    and field separator.
+
+    The returned dataset includes samples, each of which can either be a list of text fields
+    if field_separator is specified, or otherwise a single string segment produced by the
+    sample_splitter.
+
+    Parameters
+    ----------
+    filename : str or list of str
+        Path to the input text file or list of paths to the input text files.
+    encoding : str, default 'utf8'
+        File encoding format.
+    sample_splitter : function, default str.splitlines
+        A function that splits the dataset string into samples.
+    field_separator : function or None, default Splitter('\t')
+        A function that splits each sample string into list of text fields.
+        If None, raw samples are returned according to `sample_splitter`.
+    num_discard_samples : int, default 0
+        Number of samples discarded at the head of the first file.
+    field_indices : list of int, default None
+        If set, for each sample, only fields with provided indices are selected as the output.
+        Otherwise all fields are returned.
+
+    Example::
```

Comment:
I'm enabling doctest in #402.

Reply:
The api takes a path to the file as the input. To run this we have to create a file. Showing the code for file creation is verbose... Maybe we don't want to enforce all examples to be verified by doctest. wdyt

Reply:
You can either check a small file in, or have it hosted and downloaded. I'm doing the same for sentencepiece tokenizer where the tokenizer requires a bpe model file

Reply:
+1 for doctesting.